client/core/types.go

```diff
@@ -360,19 +362,71 @@ func coreOrderFromTrade(ord order.Order, metaData *db.OrderMetaData) *Order {
 
 // Market is market info.
 type Market struct {
-	Name                string   `json:"name"`
-	BaseID              uint32   `json:"baseid"`
-	BaseSymbol          string   `json:"basesymbol"`
-	QuoteID             uint32   `json:"quoteid"`
-	QuoteSymbol         string   `json:"quotesymbol"`
-	EpochLen            uint64   `json:"epochlen"`
-	StartEpoch          uint64   `json:"startepoch"`
-	MarketBuyBuffer     float64  `json:"buybuffer"`
-	Orders              []*Order `json:"orders"`
-	BaseOrderLocked     uint64   `json:"baseorderlocked"`
-	QuoteOrderLocked    uint64   `json:"quoteorderlocked"`
-	BaseContractLocked  uint64   `json:"basecontractlocked"`
-	QuoteContractLocked uint64   `json:"quotecontractlocked"`
```

Comment:
These fields have been replaced by methods. Breaking change, obviously.

Reply:
Just thinking about the rpc server and other places where we marshal this before sending it, as opposed to Go consumers that can call the method... is their a net benefit from excluding the fields?

I'm looking at `sideContractLocked`, and because the computed value never changes once a `*Market` is generated, why not stick with fields? AFAICT, the `core.Order`s generated for this `*Market` are immutable, assuming it's only coming from the `(*dexConnection).marketMap` method.

Reply:
I guess the counter-argument is that the data is already there and this sort of "summary field" is just a maintenance item. And while you're correct that we do not mutate the Go types returned through the `Core` frontend, in our javascript app that's exactly what we do to the decoded structs, so these fields are outdated with the first order or match update.

Based on that, I'm guessing a Go app which made ample use of this API might employ the same technique to keep everything synchronized as order/match updates come in. This is in fact the exact thing we were doing internally (until this PR) to avoid regenerating the entire `User` struct too much. But that optimization became unnecessary, mostly because the client was doing the struct updates anyway, so wasn't requesting the `User` as often. 

I guess my point is that even though we don't mutate these structs in Go, the caller may want to, in which case these fields could be confusing. These are the only fields we have whose accuracy relies on the immutability of the associated matches, so my preference is to remove them. 

That said, I do recognize that this leaves in the cold anyone who was consuming these via JSON conversion to another language (e.g. us). They would have to reimplement the methods themselves. If you think that's a strong enough reason to keep them, I can go back to fields.

Reply:
That all makes sense.  I'm not aware of the details of how Go consumers actually use the Users struct (do they really hold the whole `User` struct modify it in response to order notes or do they just extract `Orders`?), but my feeling is that the consumer should treat the struct as a snapshot and thus the locked amounts that derive from current order-match statuses would change as the matches change.  Yet I completely get your point of view that derived fields might as well be replaced with the functions that derive them.

But my initial comment was mainly considering someone doing `dexcctl exchanges` (or GET the analogous http api endpoint, with no interest in websocket notifications) to just get a snapshot of how much they have locked in these states, which would now require going through all the orders returned.

Reply:
Pinging some users real quick...

Reply:
OK, so the wallets RPC will still provide this info, along with `BalanceNote`s.  I'm fine with this change.